src/main/java/org/opensearch/agent/job/IndexSummaryEmbeddingJob.java

```diff
@@ -0,0 +1,286 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.opensearch.ExceptionsHelper;
+import org.opensearch.ResourceAlreadyExistsException;
+import org.opensearch.action.bulk.BulkRequest;
+import org.opensearch.action.delete.DeleteRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.agent.indices.IndicesHelper;
+import org.opensearch.client.Client;
+import org.opensearch.client.Requests;
+import org.opensearch.cluster.metadata.ComposableIndexTemplate;
+import org.opensearch.cluster.metadata.DataStream;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.metadata.IndexTemplateMetadata;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.action.ActionFuture;
+import org.opensearch.common.regex.Regex;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.index.Index;
+import org.opensearch.core.xcontent.MediaTypeRegistry;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.query.MatchAllQueryBuilder;
+import org.opensearch.ml.common.output.model.ModelTensorOutput;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.hash.Hashing;
+
+import lombok.Builder;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class IndexSummaryEmbeddingJob implements Runnable {
+
+    private Client client;
+    private ClusterService clusterService;
+    private IndicesHelper indicesHelper;
+    private MLClients mlClients;
+
+    public static String INDEX_SUMMARY_EMBEDDING_INDEX = ".index_summary_embedding_index";
+    public static String INDEX_SUMMARY_EMBEDDING_FIELD_PREFIX = "index_summary_embedding";
+    public static String INDEX_NAME_FIELD = "index_name";
+    public static String DATA_STREAM_FIELD = "data_stream";
+    public static String INDEX_PATTERNS_FIELD = "index_patterns";
+    public static String ALIAS_FIELD = "alias";
+    public static String INDEX_SUMMARY_FIELD = "index_summary";
+
+    public static String INDEX_NAME = "index";
+    public static String DATA_STREAM = "data_stream";
+    public static String ALIASES = "aliases";
+    public static String INDEX_PATTERNS = "index_patterns";
+    public static String INDEX_SUMMARY = "summary";
+    public static String INDEX_EMBEDDING = "embedding";
+    public static String SENTENCE_EMBEDDING = "sentence_embedding";
+    public static int DEFAULT_TIMEOUT_SECOND = 30;
```

Comment:
Just FYI, when I tested it locally on my mac, with 60-70 indexes, 30 seconds seems not enough.